### DISCUSSION TITLE
increase Serial buffer (addMemoryForRead) = reboots problem

### DISCUSSION THREAD

**Discussion by 666mille** [tsandmann/freertos-teensy#24] [Q&A] [answered]
Hi all,
I would to increase the serial receive buffer with the function "addMemoryForRead". But it results in a reboot loop. Does anyone have an idea?
kind regards
Holger

**Comment by tsandmann** [comment, marked as answer]
Hi Holger, 

Could you provide some sample code where this reboot loop happens? I couldn't reproduce such behaviour.

Kind regards,
Timo

**Comment by 666mille** [comment]
Hi Timo,
thanks for the quick reply. I was sick for a few days...but now an update.
You're right, the function works perfectly.
The problem was mine. I get a JSON string over 1000 bytes and I read it using the "Serial.readString" function and then passed it incorrectly to a display buffer.
With “Serial.readBytes” everything works wonderfully now!
Thank you!
